user/deployment/npm.md

```diff
@@ -132,3 +132,38 @@ after_deploy:
   - ./after_deploy_1.sh
   - ./after_deploy_2.sh
 ```
+
+### Troubleshooting
+
+Did you expect a job to publish a package but it did not? Here are some common reasons that can happen and how to resolve them.
```

Comment:
I disagree, as I have spent >1hr of my time on the first two issues, and far less than that documenting them. My intention is purely noble - spare others the grief of hunting down the cause and resolution of their misfortune. But I intentionally configured this PR with "Allow edits from maintainers" so you can do whatever you want with it.